### PR TITLE
INT-385: Ensuring the SearchParameter used by the CPS via the code. 

### PR DESCRIPTION
**Pull request by JorisHeadease**
This is added when the CPS is enabled in the config. It performs via a conditional create. Removed the "init" scripts, and reconfigured to docker-compose file to add a health-check service for HAPI, so that docker services and also depend on a "live" FHIR server.

### CODE REVIEW

File: orchestrator/careplanservice/service.go [hunk 1]
```diff
@@ -443,3 +445,51 @@ func (s Service) baseUrl() *url.URL {
...
 	return strings.Split(resourcePath, "/")[0]
 }
+
+func (s *Service) ensureSearchParameterExists() {
+	searchParamID := "CarePlan-subject-identifier"
+
+	// Define the SearchParameter
+	searchParam := fhir.SearchParameter{
+		Id:          &searchParamID,
+		Url:         "http://zorgbijjou.nl/SearchParameter/CarePlan-subject-identifier",
```
Comment (reinkrul) [inline orchestrator/careplanservice/service.go:455]: out of scope for this change, but we should probably change this to Shared Care Planning.

**Review by reinkrul — APPROVED** [review]

File: orchestrator/careplanservice/service.go (hunk 1 above)
Comment (dwelman) [inline orchestrator/careplanservice/service.go:455]: I feel like we are limiting ourselves by hardcoding these endpoints and email addresses for future testing, environments, and scaling. Should this not be configurable?

Comment (JorisHeadease) [reply orchestrator/careplanservice/service.go:455]: We could, but in FHIR it's "just" a [canonical identifier](https://www.hl7.org/fhir/r4/references.html#canonical), so they may/should not change over time. It's also just created once, when the server starts up and it's not present yet.

Comment (dwelman) [reply orchestrator/careplanservice/service.go:450]: Ok, sounds good to me

**Pull request merged**